### PR TITLE
MP-2689: Select and buy merchant product

### PR DESCRIPTION
**Pull request by tobi812**
## PR Description

IG for selecting and buying merchant products

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-feature-integration.md [hunk 1]
```diff
@@ -690,7 +690,40 @@ Make sure when you add to cart merchant product, it has `merchantReference` set.
...
+     * @return \Spryker\Zed\PublisherExtension\Dependency\Plugin\PublisherPluginInterface[]
+     */
+    protected function getPublisherPlugins(): array
+    {
+        return [
+            new MerchantProductWritePublisherPlugin(),
+            new MerchantUpdatePublisherPlugin(),
+        ];
+    }
+}
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-feature-integration.md:725]: why in Frontend section? These plugins are already presented in this IG, please move them to the corresponding place according to https://spryker.atlassian.net/wiki/spaces/DOCS/pages/478118714/Feature+integration+guides

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md [hunk 2]
```diff
@@ -1171,6 +1175,9 @@ namespace Pyz\Client\MerchantProductOfferStorage;
 use Spryker\Client\MerchantProductOfferStorage\MerchantProductOfferStorageDependencyProvider as SprykerMerchantProductOfferStorageDependencyProvider;
 use Spryker\Client\MerchantProductOfferStorage\Plugin\MerchantProductOfferStorage\DefaultProductOfferReferenceStrategyPlugin;
 use Spryker\Client\MerchantProductOfferStorage\Plugin\MerchantProductOfferStorage\ProductOfferReferenceStrategyPlugin;
+use Spryker\Client\MerchantProductStorage\Plugin\MerchantProductOfferStorage\MerchantProductProductOfferReferenceStrategyPlugin;
+use Spryker\Client\PriceProductOfferStorage\Plugin\PriceProductStorage\LowestPriceProductOfferStorageCollectionSorterPlugin;
+use Spryker\Client\PriceProductOfferStorage\Plugin\PriceProductStorage\PriceProductOfferStorageExpanderPlugin;
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md:1180]: these plugins are not related to this IG

**Review by ilyakubanov — COMMENTED** [review]
- In Prerequisites section please adjust links to IGs instead of Git repos
- MerchantProductOfferStorageDependencyProvider.php is inside of another tab here, please check formatting 
- MerchantProductPreAddToCartPlugin must be placed in Product + Cart IG

Please carefully recheck other plugins in related IGs and be sure that all from https://github.com/spryker/suite-nonsplit/pull/4296/files are presented in corresponding IGs

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md [hunk 3]
```diff
@@ -1383,6 +1391,8 @@ merchant_product_offer.sold_by,Sold by,en_US
 merchant_product_offer.sold_by,Verkauft durch,de_DE
 merchant.sold_by,Sold by,en_US
 merchant.sold_by,Verkauft durch,de_DE
+merchant_product.sold_by,Sold by,en_US
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md:1394]: This one translationes seems should not be hear, cos used in MerchantProductWidget module, so need to be the part of Marketplace Product feature IG

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md [hunk 4]
```diff
@@ -1181,6 +1188,7 @@ class MerchantProductOfferStorageDependencyProvider extends SprykerMerchantProdu
     {
         return [
             new ProductOfferReferenceStrategyPlugin(),
+            new MerchantProductProductOfferReferenceStrategyPlugin(),
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md:1191]: As far as this one plugin is registered in `MerchantProductStorage` module, which is a part of `MerchantProduct` feature. I propose to move it there and provide a link in this IG for the case when customer wants to install `Product Offer` feature after `Merchant Product` feature

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-feature-integration.md [hunk 5]
```diff
@@ -335,13 +335,41 @@ class ProductStorageDependencyProvider extends SprykerProductStorageDependencyPr
...
+{
+    /**
+     * @return \Spryker\Client\MerchantProductOfferStorageExtension\Dependency\Plugin\ProductOfferReferenceStrategyPluginInterface[]
+     */
+    protected function getProductOfferReferenceStrategyPlugins(): array
+    {
+        return [
+             new MerchantProductProductOfferReferenceStrategyPlugin(),
+        ];
+    }
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-feature-integration.md:369]: must be placed in Marketplace Product + Marketplace Product Offer

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md (hunk 4 above)
Comment (ilyakubanov) [reply docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md]: must be placed in Marketplace Product + Marketplace Product Offer

File: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md (hunk 3 above)
Comment (ilyakubanov) [reply docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md]: please move to the right place
